extensions/narayana-jta/deployment/src/main/java/io/quarkus/narayana/jta/deployment/NarayanaJtaProcessor.java

```diff
@@ -96,6 +114,40 @@ public void build(NarayanaJtaRecorder recorder,
         recorder.setDefaultTimeout(transactions);
     }
 
+    @BuildStep
+    void testTx(LaunchModeBuildItem lm, BuildProducer<GeneratedBeanBuildItem> generatedBeanBuildItemBuildProducer) {
```

Comment:
Hm, so you generate an interceptor class because you can't add the `@TestTransaction` interceptor binding, right? Ok, that's probably the simplest solution. However, I'd move as much as possible logic to the parent class, i.e. injection field of `UserTransaction` and the whole `@AroundInvoke` method. If it doesn't work then we have a bug in ArC...

Reply:
It didn't work, although I did not have time to look into why.

Reply:
Maybe it's because the parent class is not part of the index?